components/AdminUserSettings/AdminUserSettings.tsx

```diff
@@ -1,25 +1,75 @@
 // Copyright (c) Gridiron Survivor.
 // Licensed under the MIT License.
 
+'use client';
+import React, { JSX } from 'react';
+import { Button } from '../Button/Button';
+import { useDataStore } from '@/store/dataStore';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '../TableDropDownMenu/TableDropDownMenu';
 import { LucideChevronsUpDown } from 'lucide-react';
-import React from 'react';
+import { useAuthContext } from '@/context/AuthContextProvider';
+import { useRouter } from 'next/navigation';
+import LinkCustom from '../LinkCustom/LinkCustom';
 
 /**
- * The admin user settings component.
- * @returns The rendered admin user settings.
+ * Renders admin user settings.
+ * @returns {JSX.Element} The rendered admin user settings component.
  */
-export const AdminUserSettings = (): React.JSX.Element => {
+export const AdminUserSettings = (): JSX.Element => {
+  const router = useRouter();
+  const { logoutAccount } = useAuthContext();
+  const { user } = useDataStore((state) => state);
+
+  /**
+   * Handles the logout.
+   * @returns {Promise<void>} The logout promise.
+   */
+  const handleLogout = async (): Promise<void> => {
+    try {
+      await logoutAccount();
+      router.push('/login');
+    } catch (error) {
+      throw new Error('Logout failed');
+    }
+  };
+
   return (
-    <div
-      className="admin-user-settings flex gap-2 px-2 py-2 items-center outline outline-border rounded text-foreground"
-      data-testid="admin-user-settings"
-    >
-      <span className="bg-cyan-500 w-8 h-8 rounded-full" />
-      <p>Users Name</p>
-      <LucideChevronsUpDown
-        className="ml-auto text-muted-foreground"
-        size={16}
-      />
-    </div>
+    <DropdownMenu>
+      <DropdownMenuTrigger
+        className="w-56 focus:outline-none mx-auto"
+        data-testid="admin-user-settings"
+      >
+        <div className="admin-user-settings w-full flex space-between gap-2 px-2 py-2 items-center border border-border rounded-lg text-foreground overflow-hidden">
+          <span className="bg-cyan-500 w-8 h-8 rounded-full" />
+          <p className="truncate ... w-36 ">{user.email}</p>
+          <LucideChevronsUpDown className="text-muted-foreground" size={16} />
+        </div>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="border border-border rounded-lg p-0 w-56 ">
+        <DropdownMenuItem className="cursor-pointer rounded-b-none flex focus:bg-muted">
+          <LinkCustom
```

Comment:
Why is this a link and sign out a button? They should be the same.

Reply:
@shashilo they are not the same.  The button runs a logout script.  The link is a route to a page.  Links should not be buttons.

Reply:
I'll make changes to this

Reply:
I get an error for LinkCustom props when adding a onClick prop. should I change this in the component?

Reply:
If there's no onClick handler, we should add one to the component